docs/README.md

```diff
@@ -195,6 +214,220 @@ sudo ip link del wg0
 
 Since the wireguard agents are userspace in both Windows and Darwin, the tunnel interface is removed when the agent process exits.
 
+### Deploying on Kubernetes managed Node
+
+Instructions mentioned in [Deploying on Node](#deploying-on-node) can be used here to deploy the Apex agent on Kubernetes-managed nodes. However, deploying the agent across all the nodes in a sizable Kubernetes cluster can be a challenging task. The following section provides instructions to deploy Kubernetes style manifest to automate the deployment process.
+
+#### Setup the configuration
+
+Agent deployment in Kubernetes requires a few initial configuration details to successfully deploy the agent and onboard the node. This configuration is provided through `kustomization.yaml`. Make a copy of the sample `kustomization.yaml.sample`](../deploy/apex-client/overlays/dev/kustomization.yaml.sample) and rename it to `kustomization.yaml`.
+
+Fetch the CA cert from the Kubernetes cluster that is running the Apex controller, and set the return cert string to `cert` literal of `secretGenerator` in `kustomization.yaml`.
+
+```sh
+kubectl get secret -n apex apex-ca-key-pair -o json | jq -r '.data."ca.crt"'
+```
+
+Update the `<APEX_CONTROLLER_IP>` with the IP address where the Apex controller is reachable. If you are running the Apex stack in Kind cluster on your local machine, set it to the public IP of the machine.
+
+**Note**
+Current username and password are default configuration for development environment, which is very likely to change in near future.
+
+The Following is an example `kustomization.yaml`:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: apex
+bases:
+  - ../../base
+secretGenerator:
+- name: apex-client-secret
+  literals:
+  - cert=<APEX_CONTROLLER_CERT>
+  - url=https://apex.local
+  - auth_url=https://auth.apex.local/token
+  - username=kitteh1
+  - password=floofykittens
+commonLabels:
+  app.kubernetes.io/component: apex-client
+  app.kubernetes.io/instance: apex-client
+  app.kubernetes.io/name: apex-client
+patchesJson6902:
+  - target:
+      kind: DaemonSet
+      name: apex
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/hostAliases/0/ip
+        value: 54.65.23.108
+patchesStrategicMerge:
+  - node_selector.yaml
+```
```

Comment:
I wonder if we should link to `kustomization.yaml.sample` in the tree to help avoid this and the sample file getting out of sync over time?

Reply:
Yup, makes sense. Updated the readme.